### ISSUE TITLE
`math.evaluate("f(x,x)=x")` should throw an error

### ISSUE THREAD

**Issue by josdejong** [josdejong/mathjs#2818]
### Discussed in https://github.com/josdejong/mathjs/discussions/2815

<div type='discussions-op-text'>

<sup>Originally posted by **ChristopherChudzicki** October 16, 2022</sup>
**Describe the bug**
`f(x,x) = x` parses and evaluates to a function with somewhat unexpected behavior.

**To Reproduce**
Run:
```js
const math = require('mathjs')
const f = math.evaluate("f(x, x) = x")
f(1, 2)
```

I came across this while working on (the rewrite of) my app, math3d.org. See https://www.math3d.org/S9Kp3MDh5 for how this came up. I want user input of `f(x, x)` to be an error. It seems like something unintended/confusing by users. Certainly I can handle this as a consumer of the mathjs library.

But... how should MathJs handle this? Options:

1. The current behavior is correct: allow parsing and evluating `f(x, x) = ...`, with later parameter values taking precedence.
2. Throw an error during parsing
3. Throw an error during evaluation: can't determine which value of x is correct. 

(2) Seems too restrictive. Parsing the expression can be useful for getting structural information.  (1) and (3) both seem somewhat reasonable... Curious if this has come up before / is intended behavior. </div>

**Comment by josdejong** [comment]
This is a bug, we should throw an error when entering the same variable name multiple times.

**Comment by josdejong** [comment]
I've created a fix in #2819.

**Comment by josdejong** [comment]
Fix published now in `v11.3.1`
